src/Builder/Query.php

```diff
@@ -40,6 +44,22 @@ public static function regex(Regex|string $regex, string|null $flags = null): Re
         return self::generatedRegex($regex);
     }
 
+    /**
+     * The $elemMatch operator matches documents that contain an array field with at least one element that matches all the specified query criteria.
+     *
+     * @see https://www.mongodb.com/docs/manual/reference/operator/query/elemMatch/
+     */
+    public static function elemMatch(QueryInterface|FieldQueryInterface|Type|stdClass|array|bool|float|int|string|null ...$query): ElemMatchOperator
+    {
+        // $elemMatch accepts field query without field name ...
```

Comment:
@jmikola what do you thing of this comment:
```suggestion
        // To match elements in non-document arrays, $elemMatch can be used with field query operators,
        // They must be passed as a list of operators without a field name as key.
```

Reply:
Consider:

> $elemMatch can be used to match arrays of documents and non-document values alike. When matching non-document values, field query operators should be passed as a list of operators without the field name as a key.

Alternatively, you can replace `field query operators` with the class/interface name.

I assume this means users _cannot_ pass an associative array of the raw query operators, since that might be interpreted as the API for matching an array of documents?

```
[ '$gt' => 5, '$lt' => 10 ]
```

Might that be interpreted as a equality matches on "$gt" and "$lt" fields paths?

Reply:
You're right.
I'm afraid that given the ambiguity, we have no choice but to remove the named arguments and make the use of a single `Query` or `FieldQuery` argument, which will be created with the `Query::query` or `Query::fieldQuery` function.